polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -491,78 +491,71 @@ void MemorySelector::analyze(LoopLikeOpInterface loop, AccessKind accessKind) {
       continue;
     }
 
-    // Get the inter-thread access pattern and classify the memory access.
+    memRefAccessToMemSpace[memRef] =
+        selectMemorySpace(*memAccess, threadVars, accessKind);
+
+    LLVM_DEBUG({
+      if (memRefAccessToMemSpace.at(memRef) == MemorySpace::Shared)
+        llvm::dbgs().indent(2) << "shared memory space\n";
+      else {
+        assert(memRefAccessToMemSpace.at(memRef) == MemorySpace::Global);
```

Comment:
now that you may return MemorySpace::Texture, can you add a print method for MemorySpace, and use that here?

Reply:
done